### ISSUE TITLE
Documentation for DiceLoss is misleading

### ISSUE THREAD

**Issue by ebrahimebrahim** [Project-MONAI/MONAI#3306]
The [documentation for `DiceLoss`](https://docs.monai.io/en/latest/losses.html?highlight=dice%20loss#diceloss) says (my bold):

> Compute average Dice loss between two tensors. It can support both multi-classes and multi-labels tasks. Input **logits** _input_ (BNHW[D] where N is number of classes) is compared with ground truth _target_ (BNHW[D]). Axis N of _input_ is expected to have **logit** predictions for each class rather than being image channels, while the same axis of target can be 1 or N (one-hot format)...

The term "logit" here doesn't seem to match how `DiceLoss` [actually works](https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/dice.py#L162-L176); it expects class probabilities and not logit values. That can change for one of the inputs if the parameters `sigmoid` or `softmax` are set to `True`, but those parameters are `False` by default.

**Comment by Nic-Ma** [comment]
Hi @ebrahimebrahim ,

Thanks for your feedback.
I will try to double-check the related document ASAP.

Thanks.
